src/OpenTelemetry/DistributedContext/DistributedContextState.cs

```diff
@@ -13,17 +13,24 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 // </copyright>
-using System.Collections.Generic;
+
+using System;
 
 namespace OpenTelemetry.Context
 {
-    public sealed class NoopTagContext : TagContextBase
+    internal struct DistributedContextState : IDisposable
     {
-        internal static readonly ITagContext Instance = new NoopTagContext();
+        private DistributedContext context;
+
+        internal DistributedContextState(in DistributedContext context)
+        {
+            this.context = AsyncLocalDistributedContextCarrier.Instance.Current;
+            ((AsyncLocalDistributedContextCarrier)AsyncLocalDistributedContextCarrier.Instance).OverwriteCurrent(in context);
```

Comment:
so values are not inherited? It may be an unexpected behavior for scope API

Reply:
what you mean by values are not inherited? could you give some example what would be broken by that?

Reply:
if i defined a larger scope with the value "appVersion=1.0" and later smaller scope with the property "flightID=Foo" I want both properties to be propagated downstream. Without manually merging these two collections

Reply:
Is this always the case? I mean when setting smaller scope, should it merge it with the current larger scope? or this should be be opt-in behavior? I am seeing this is as implementation details but would be easy to support it.

Reply:
scenario wise - it is expected to inherit scopes. But there should be a mechanism to 

1. wipe it and start fresh (typical for situation you enter some "hostile" code that will call into external services)  
2. remove/overwrite certain keys

Reply:
> wipe it and start fresh (typical for situation you enter some "hostile" code that will call into external services)

This can be done easily by 
``C#
DistributedContext.SetCurrent(DistributedContext.Empty);
```

> remove/overwrite certain keys

Either can be done manually by getting the DistributedContext.Entries and then add/remove/modify. Or we can add some helper methods inside the DistributedContext to do that.

Reply:
https://github.com/open-telemetry/opentelemetry-dotnet/issues/374